test/unit-test/test_simulator.c

```diff
@@ -0,0 +1,135 @@
+// Copyright 2023-2024 Shift Crypto AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "hww.h"
+#include "memory/bitbox02_smarteeprom.h"
+#include "usb/usb_packet.c"
+#include "usb/usb_processing.c"
+#include "usb/usb_processing.h"
+#include "workflow/idle_workflow.h"
+#include <fcntl.h>
+#include <memory/memory.h>
+#include <mock_memory.h>
+#include <queue.h>
+#include <random.h>
+#include <rust/rust.h>
+#include <sd.h>
+#include <stdio.h>
+#include <unistd.h>
+
+#include <netdb.h>
+#include <netinet/in.h>
+#include <sys/socket.h>
+
+#define BUFFER_SIZE 1024
+
+int data_len;
+int sockfd;
```

Comment:
make `sockfd*` an argument to the get_usb_message_socket function, then you don't need a global sockfd variable and it can be local. 

looks like we can't do much about data_len being global for now, that seems it would require some refactoring of the usb processing functions, which we don't need to go into now :smile: